joblib/parallel.py

```diff
@@ -54,7 +54,7 @@
 # pool that does not tolerate forking
 if hasattr(mp, 'get_start_method'):
     method = os.environ.get('JOBLIB_START_METHOD')
-    if (method is None and mp.get_start_method() == 'fork'
+    if (method == 'forkserver'  and mp.get_start_method() == 'fork'
             and 'forkserver' in mp.get_all_start_methods()):
         method = 'forkserver'
     DEFAULT_MP_CONTEXT = mp.get_context(method=method)
```

Comment:
Actually I think that if `JOBLIB_START_METHOD` is set we should just use it and otherwise just use `DEFAULT_MP_CONTEXT= mp.get_context()` which is the `fork` context by default under POSIX and spawn under windows.